launcher/JavaCommon.cpp

```diff
@@ -17,6 +17,17 @@ bool JavaCommon::checkJVMArgs(QString jvmargs, QWidget *parent)
             QMessageBox::Warning)->exec();
         return false;
     }
+    // block lunacy with passing required version to the JVM
+    if (jvmargs.contains(QRegExp("-version:.*"))) {
+        auto warnStr = QObject::tr(
+            "You tried to pass required java version argument to the JVM (using \"-version=xxx\"). This is not safe and will not be allowed.\n"
+            "This message will be displayed until you remove this from the JVM arguments.");
```

Comment:
If a user tries to use the `-version:` flag, they're probably trying to select an appropriate Java version. The message would be more helpful if it suggested manually changing the Java version in the instance settings to accomplish this instead.

Reply:
I mean yeah I just backported that patch from upstream lol

Reply:
I think it's fine. I have read that some people use this argument to "improve" their Java options or whatever. But it seems like it causes issues, which is why upstream blocks it.